server/controllers/userController.js

```diff
@@ -0,0 +1,87 @@
+const asyncHandler = require('express-async-handler');
+const {  validationResult } = require('express-validator');
+const {assignToken} = require('../utils/token');
+const bcrypt = require('bcryptjs');
+const User = require('../models/User');
+
+
+//@route    POST /users/register
+//@des      register a new user
+//@access   public
+const registerUser = asyncHandler(async(req,res)=>{
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(400).json({ errors: errors.array() });
+    }
+    let {name, email, password,image} = req.body;
+    
+    const user = await User.findOne({email});
+    
+    if(user){
+        res.status(400)
+        throw new Error('user already exist')
+    }
+    
+    const newUser = new User({name,email,password,image});
+    await newUser.save();
+
+    token = assignToken(newUser.id);
+
+    res.cookie('authToken',`Bearer ${token}`,{maxAge:3600, httpOnly:true});
```

Comment:
If we're storing the token as an httpOnly cookied, I don't believe we need `Bearer` for anything